### PR TITLE
Final step of C++ cleanup

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: src/eei.h [hunk 1]
```diff
@@ -50,8 +51,22 @@ class InternalErrorException : std::exception {
...
 
+struct ExecutionResult {
+  uint64_t gasLeft;
+  std::vector<char> returnValue;
+};
+
 struct EthereumInterface : ShellExternalInterface {
-  EthereumInterface(Hera const& _hera, HeraCall & _call) : ShellExternalInterface(), hera(_hera), call(_call) { (void)hera; }
+  EthereumInterface(
+    struct evm_context const& _context,
```
Comment (chfast) [inline src/eei.h]: You don't need `struct` prefix in C++.

File: src/eei.h [hunk 2]
```diff
@@ -50,8 +51,22 @@ class InternalErrorException : std::exception {
...
+    struct evm_message const& _msg,
+    ExecutionResult & _result
+  ):
+    ShellExternalInterface(),
+    context(_context),
+    msg(_msg),
+    result(_result)
+  { (void)context; }
```
Comment (chfast) [inline src/eei.h]: The `(void)` is not needed.

File: src/hera.cpp [hunk 3]
```diff
@@ -109,7 +160,7 @@ static void evm_destroy(struct evm_instance* instance)
   free(instance);
 }
 
-struct evm_instance* hera_create()
+struct evm_instance* hera_create(void)
```
Comment (chfast) [inline src/hera.cpp]: The `void` is only needed in declarations.

File: src/eei.h (hunk 2 above)
Comment (axic) [reply src/eei.h]: Needed to silence unused variable warning (soon it will be used).

**Pull request merged**